### PR TITLE
[Reformer] Make random seed generator available on random seed and not on model device

### PR DESCRIPTION
**Pull request by patrickvonplaten**
@LysandreJik - that might possibly fix the circle ci error.

### CODE REVIEW

File: src/transformers/modeling_reformer.py [hunk 1]
```diff
@@ -1417,17 +1418,18 @@ def _init_feed_forward_seed(self):
             call and 1 forward call in backward
             to recalculate activations.
         """
-
+        #
```
Comment (patrickvonplaten) [inline src/transformers/modeling_reformer.py:1421]:
```suggestion
        # randomize seeds
        # use cuda generator if available
```

File: src/transformers/modeling_reformer.py [hunk 2]
```diff
@@ -1417,17 +1418,18 @@ def _init_feed_forward_seed(self):
...
         # randomize seeds
```
Comment (patrickvonplaten) [inline src/transformers/modeling_reformer.py:1422]:
```suggestion
```

File: src/transformers/modeling_reformer.py [hunk 3]
```diff
@@ -1417,17 +1418,18 @@ def _init_feed_forward_seed(self):
             call and 1 forward call in backward
             to recalculate activations.
         """
-
         # randomize seeds
-        if next(self.parameters()).device.type == "cuda":
+        # use cuda generator if available
```
Comment (patrickvonplaten) [inline src/transformers/modeling_reformer.py:1422]:
```suggestion
        # randomize seeds
```

File: src/transformers/modeling_reformer.py [hunk 4]
```diff
@@ -1417,17 +1418,18 @@ def _init_feed_forward_seed(self):
...
+        # use cuda generator if available
```
Comment (patrickvonplaten) [inline src/transformers/modeling_reformer.py:1423]:
```suggestion
```

**Review by LysandreJik — APPROVED** [review]
Tested on the machine, it works flawlessly! Thanks!

**Pull request merged**